ghostwriter/static/js/tinymce/plugins/ghostwriter-comments/plugin.min.js

```diff
@@ -0,0 +1,110 @@
+tinymce.PluginManager.add('ghostwriter-comments', function(editor) {
+    // Function to decode the users hex username into ascii
+    function hex2ascii(hexString) {
+        let decodedStr = '';
+        for (let i = 0; i < hexString.length; i += 2) {
+            const byte = parseInt(hexString.substr(i, 2), 16);
+            decodedStr += String.fromCharCode(byte);
+        }
+        return decodedStr
+    }
+
+    // Create Comment
+    function createComment(comment, selection) {
+        const commentNode = editor.selection.getNode();
+        let commentThread = readComments(commentNode);
+        commentThread.push({ author: username, comment: comment });
```

Comment:
I'm curious as to where the `username` variable is set, and also why it's in hex. We should be able to just provide it as a normal string. I'm also not sure if it will be set in the admin pages. I'll look into it.

Reply:
The username is a constant set on the `base_generic.html` and `base_generic_empty.html` template so I think it should be everywhere (maybe...?)

Reply:
Yeah, found it.

It's not provided if a user isn't logged in, but we don't have any rich text fields for anonymous users, so whatever.

Admin pages won't use those templates, though I don't think we need to have comments anywhere in there, so we can probably just disable comments if there's no username.

Seems like the section it's defined in is setting up the websocket notifiers, and the variables it uses are just leaked to the global scope, which I'm not a fan of. But that's a different issue.